### PR TITLE
Update cmake-format and automatically sort file lists

### PR DESCRIPTION
**Pull request by dragomirecky**
This PR
a) updates cmake-format to version 0.6.13
b) configures it to automatically sort file lists within `target_sources` calls
c) reformats everything

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,11 +1,10 @@
 # See https://pre-commit.com for more information
 # See https://pre-commit.com/hooks.html for more hooks
 repos:
-- repo: https://github.com/dragomirecky/cmake-format-pre-commit-hook
-  rev: 'v0.6.0'
+- repo: https://github.com/cheshirekow/cmake-format-precommit
```
[the remark this answers is not in the document]
Comment (dragomirecky) [reply .pre-commit-config.yaml:4]: The repo is maintained by the author of cmake-format, so we are not loosing any control here.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
+  rev: v0.6.13
   hooks:
-    - id: cmake-format  # cmake formatter
-      files: (^|/)(CMakeLists.*|.*\.cmake|.*\.cmake.in)$
+    - id: cmake-format
```
Comment (DRracer) [inline .pre-commit-config.yaml:7]: the files pattern, which has been here before, is no longer needed with the new cmake-format?

File: lib/Arduino_Core_Buddy/CMakeLists.txt [hunk 3]
```diff
@@ -21,13 +21,8 @@ target_include_directories(
...
 
 target_compile_definitions(
-  CoreBuddy
-  PUBLIC
-  ARDUINO=187
-  $<$<STREQUAL:${ARDUINO_CORE_TARGET},STM32F4xx>:STM32F4xx>
-  HAVE_HWSERIAL2
-  USBD_USE_CDC
-  USBCON
+  CoreBuddy PUBLIC ARDUINO=187 $<$<STREQUAL:${ARDUINO_CORE_TARGET},STM32F4xx>:STM32F4xx>
```
Comment (DRracer) [inline lib/Arduino_Core_Buddy/CMakeLists.txt:24]: do we really want the compile definitions squashed into 2 lines?

File: src/marlin_stubs/pause/CMakeLists.txt [hunk 4]
```diff
@@ -3,12 +3,12 @@ target_sources(firmware PRIVATE G27.cpp M125.cpp M603.cpp)
 if(HAS_PAUSE)
   target_sources(
     firmware
-    PRIVATE M600.cpp
+    PRIVATE M1700_1_parse.cpp
```
Comment (DRracer) [inline src/marlin_stubs/pause/CMakeLists.txt:6]: :rofl: lexicographical ordering for numbers

File: tests/unit/lang/format_print_will_end/CMakeLists.txt [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
+  format_print_will_end_tests ../../../../src/lang/format_print_will_end.cpp
+                              ../../../../src/lang/translator.cpp tests.cpp
+  )
 
 # define required search paths
 target_include_directories(format_print_will_end_tests PUBLIC . ${CMAKE_SOURCE_DIR}/src/lang)
 
-# define custom macros
-# target_compile_definitions(format_print_will_end_tests PUBLIC)
+# define custom macros target_compile_definitions(format_print_will_end_tests PUBLIC)
```
Comment (DRracer) [inline tests/unit/lang/format_print_will_end/CMakeLists.txt:10]: Merging comments? Seriously? In this case it has its meaning not-to merge them.

Comment (dragomirecky) [reply tests/unit/lang/format_print_will_end/CMakeLists.txt]: solved by https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/2537

File: lib/Arduino_Core_Buddy/CMakeLists.txt (hunk 3 above)
Comment (dragomirecky) [reply lib/Arduino_Core_Buddy/CMakeLists.txt]: Not ideal here, I agree. But I tried a few more settings and when they improved the situation here, they broke the formatting on much more other places. So I would go with this for now.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (dragomirecky) [reply .pre-commit-config.yaml:7]: No 👍

File: src/gui/CMakeLists.txt [hunk 6]
```diff
@@ -78,12 +78,12 @@ if(HAS_SELFTEST)
...
     firmware
     PRIVATE MItem_basic_selftest.cpp
+            screen_menu_calibration.cpp
+            screen_menu_steel_sheets.cpp
             ScreenSelftest.cpp
             selftest_frame.cpp
             # steel sheets requiere first layer calibration, which require wizard, which require
             # selftest TODO add HAS_STEEL_SHEETS property
-            screen_menu_steel_sheets.cpp
-            screen_menu_calibration.cpp
```
Comment (mkbel) [inline src/gui/CMakeLists.txt:86]: Looks like files drifted away of its comment.

**Review by mkbel — APPROVED** [review]

**Pull request merged**